akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -218,8 +277,18 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
             // are enforced from the start of the connection so there's no need to invoke
             // `enforceSettings(initialLocalSettings)`
 
-            case PingFrame(true, _)  =>
-            // ignore for now (we don't send any pings)
+            case PingFrame(true, _) =>
+              pingState match {
+                case OptionVal.Some(state) =>
+                  state.pingsWithoutData += 1L
+                  if (settings.maxPingsWithoutData > 0) {
+                    // FIXME fail with specific exception or some more graceful failure?
```

Comment:
Yep, should be GOAWAY

Reply:
Now changed, however `pushGOAWAY` does not lead to the connection being torn down, I guess that is what the 
` // FIXME: handle the connection closing according to the specification` is about?